src/eth/DSProxyService.js

```diff
@@ -52,10 +52,19 @@ export default class DSProxyService extends PrivateService {
   }
 
   async ensureProxy() {
-    if (!(await this.currentProxy())) return this.build();
+    if (!(await this.currentProxy())) {
+      await this.build();
+    }
+    return await this.currentProxy();
   }
 
   async build() {
+    if (await this.currentProxy()) {
+      throw new Error(
+        'This account already has a proxy deployed at ' +
+          (await this.currentProxy())
```

Comment:
let's store this in a local variable so we don't hit the network twice

Reply:
`currentProxy` actually caches the address it gets from the network the first time, and only makes another call if the account has changed. It might be more readable with a variable anyway, though